src/Symfony/Component/Workflow/DataCollector/WorkflowDataCollector.php

```diff
@@ -65,6 +65,15 @@ public function lateCollect(): void
 
     public function getName(): string
     {
+        // Don't loop in the constructor to defer initializing workflows post-instantiation
+        foreach ($this->workflows ?? [] as $workflow) {
```

Comment:
`$workflows` cannot be `null`, it's an `iterable` and the class is `final`.
```suggestion
        foreach ($this->workflows as $workflow) {
```

Reply:
I applied this because if I remove the it, the same for the event dispatcher makes some tests fail, because uninitialized. Might be related to the way the datacollectors are serialized.